### PR TITLE
Builtin methods handle some exceptions

### LINKED ISSUE #7333: Inject fallback specialization to builtin methods
Inject `@Fallback` specializations to the builtin methods that will throw `PanicError`, rather than `UnsupportedSpecializationException`.

In the connection with IDE, Engine does not properly handle exceptions different from `PanicError` or data flow errors. We should wrap any exceptions either in `PanicError` or data flow errors.

It is easy to make a mistake when writing Truffle specialization, like in the case of [Text.parse_table engine failure](https://github.com/enso-org/enso/issues/7266#issuecomment-1634121283). The lack of `@Fallback` specialization prolonged the investigation time there.

### PR DESCRIPTION
**Pull request by Akirathan**
Closes #7333

### Pull Request Description

For a specific subset of `RuntimeException` or `Error`, the invocation builtin method is wrapped in `try {...} catch (ErrorType t) {...}` and the exception is converted to either panic or a dataflow error. So far, we can handle only `org.graalvm.truffle.api.dsl.UnsupportedSpecializationException`. All the other exception types are not caught at all.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [X] The documentation has been updated, if necessary.
- [X] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [X] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [X] Unit tests have been written where possible.
  - [X] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: engine/interpreter-dsl-test/src/test/java/org/enso/interpreter/dsl/test/ThrowableCatchTest.java
Comment (JaroslavTulach) [inline engine/interpreter-dsl-test/src/test/java/org/enso/interpreter/dsl/test/ThrowableCatchTest.java:1]: New test in `interpreter-dsl-test`. Good!

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/MethodProcessor.java [hunk 1]
```diff
@@ -341,6 +359,24 @@ private void generateCode(MethodDefinition methodDefinition) throws IOException
 
       out.println();
 
+      out.println("  /**");
+      out.println("   * Transforms a Java Throwable into either a dataflow-error or a Panic exception.");
+      out.println("   * @param t The Throwable thrown from the {@code execute} method.");
+      out.println("   */");
+      out.println("  private static Object handleThrowable(Throwable t, Node locationNode) {");
+      out.println("    CompilerDirectives.transferToInterpreterAndInvalidate();");
```
Comment (JaroslavTulach) [inline lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/MethodProcessor.java]: I am not sure we should catch and convert all `Throwable`. I know that `ProxyNode` & co. in Truffle instrumentation rely on `ThreadDeath` being propagated for _rewind frame_ functionality in debugger. Also silently swallowing `InterruptedException` might have _uninterruptible_ consequences. We should probably at least propagate `ThreadDeath`.

**Review by JaroslavTulach — APPROVED** [review]
Nice test. Consider propagating `ThreadDeath` at least.

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/MethodProcessor.java (hunk 1 above)
Comment (Akirathan) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/MethodProcessor.java]: Right. I have learned that the hard way - we should not. I have changed that to only catch some specific exceptions that we can handle at this point. For now, it's only `UnuspportedSpecializationException`. See in c4f197db392ba802dd030490bdb0d299848bf1db

**Comment by Akirathan** [comment]
Trying to catch all the `Throwable` and convert them to panics or dataflow errors is not a good idea:
- `com.oracle.truffle.api.nodes.ControlFlowException` should not be caught at all.
- Rethrowing `Throwable` would require incompatible method signature modification.

**Pull request merged**